core/kernel.go

```diff
@@ -73,300 +65,305 @@ const (
-				return process.ShutdownFunc(func(ctx context.Context) error {
-					stop := time.Now()
-					return logger.InfoMsg("Burrow is shutting down. Prepare for re-entrancy.",
-						"announce", "shutdown",
-						"shutdown_time", stop,
-						"elapsed_run_time", stop.Sub(start).String())
-				}), err
-			},
-		},
-		{
-			Name:    "RPC/info",
-			Enabled: rpcConfig.Info.Enabled,
-			Launch: func() (process.Process, error) {
-				server, err := rpcinfo.StartServer(kern.Service, "/websocket", rpcConfig.Info.ListenAddress, kern.Logger)
-				if err != nil {
-					return nil, err
-				}
-				return server, nil
-			},
-		},
-		{
-			Name:    "RPC/metrics",
-			Enabled: rpcConfig.Metrics.Enabled,
-			Launch: func() (process.Process, error) {
-				server, err := metrics.StartServer(kern.Service, rpcConfig.Metrics.MetricsPath,
-					rpcConfig.Metrics.ListenAddress, rpcConfig.Metrics.BlockSampleSize, kern.Logger)
-				if err != nil {
-					return nil, err
-				}
-				return server, nil
-			},
-		},
-		{
-			Name:    "RPC/GRPC",
-			Enabled: rpcConfig.GRPC.Enabled,
-			Launch: func() (process.Process, error) {
-				listen, err := net.Listen("tcp", rpcConfig.GRPC.ListenAddress)
-				if err != nil {
-					return nil, err
-				}
 
-				grpcServer := rpc.NewGRPCServer(kern.Logger)
-				var ks *keys.KeyStore
-				if keyStore != nil {
-					ks = keyStore
-				}
-
-				if keyConfig.GRPCServiceEnabled {
-					if keyStore == nil {
-						ks = keys.NewKeyStore(keyConfig.KeysDirectory, keyConfig.AllowBadFilePermissions)
-					}
-					keys.RegisterKeysServer(grpcServer, ks)
-				}
+	tmGenesisDoc := tendermint.DeriveGenesisDoc(&genesisDoc)
+	heightValuer := log.Valuer(func() interface{} { return kern.Blockchain.LastBlockHeight() })
+	tmLogger := kern.Logger.With(structure.CallerKey, log.Caller(LoggingCallerDepth+1))
+	tmLogger = tmLogger.With("height", heightValuer)
+	kern.Node, err = tendermint.NewNode(tmConf.TendermintConfig(), privVal, tmGenesisDoc, app, metricsProvider, nodeKey, tmLogger)
+	return err
+}
 
-				rpcquery.RegisterQueryServer(grpcServer, rpcquery.NewQueryServer(kern.State, nameRegState, proposalRegState,
-					kern.Blockchain, kern.State, nodeView, kern.Logger))
+func (kern *Kernel) LoadTransactor() (err error) {
+	nodeView, err := kern.GetNodeView()
+	if err != nil {
+		return err
+	}
 
-				rpctransact.RegisterTransactServer(grpcServer, rpctransact.NewTransactServer(kern.Transactor, txCodec))
+	accountState := kern.State
+	nameRegState := kern.State
+	kern.Service = rpc.NewService(accountState, nameRegState, kern.Blockchain, kern.State, nodeView, kern.Logger)
 
-				rpcevents.RegisterExecutionEventsServer(grpcServer, rpcevents.NewExecutionEventsServer(kern.State,
-					kern.Emitter, kern.Blockchain, kern.Logger))
+	if nodeView == nil {
+		checkTx := func(tx tmTypes.Tx, cb func(*abciTypes.Response)) error {
+			ctx := abci.TxExecutor("CheckTx", kern.exeChecker, kern.txCodec, kern.Logger.WithScope("CheckTx"))(tx)
```

Comment:
Superceded by below perhaps but let's not use `ctx` it's the idiomatic name for `context.Context` variables